### PR TITLE
Feature animation/streaming update rate

### PR DESCRIPTION
**Pull request by DavidMansolino**
Dual PR of omichel/webots#6347

### CODE REVIEW

**Comment by DavidMansolino** [comment]
👍

File: guide/web-animation.md [hunk 1]
```diff
@@ -45,13 +45,32 @@ Please refer to [this section](web-scene.md#how-to-embed-a-web-scene-in-your-web
...
 The other VRML fields are not recorded in the animation file.
 Node insertion or deletion are also not recorded in the animation file.
 
 Additional limitations are described in [this section](web-scene.md#limitations).
 
 
+### Scene refresh rate
+
+The scene refresh rate is defined by the `WorldInfo.FPS` field.
+However, the `WorldInfo.basicTimeStep` field have an impact on the refresh rate too, because the update are only sent during the simulation step. The actual refresh rate can be computed with the following formula:
```
Comment (stefaniapedrazzi) [inline guide/web-animation.md]: update are only sent -> update**s** are only sent

File: guide/web-animation.md [hunk 2]
```diff
@@ -45,13 +45,32 @@ Please refer to [this section](web-scene.md#how-to-embed-a-web-scene-in-your-web
...
+
+%figure "Animation refresh rate."
```
Comment (stefaniapedrazzi) [inline guide/web-animation.md]: to me it looks a little bit strange to have a caption for a formula

**Comment by DavidMansolino** [comment]
> rate = 1000 / min(step * ceil(1000 / (step * FPS)), step)

Shouldn't it be `rate = 1000 / min(step / ceil(1000 / (step * FPS)), step)` instead?

**Comment by stefaniapedrazzi** [comment]
> Shouldn't it be rate = 1000 / min(step / ceil(1000 / (step * FPS)), step) instead?

No, `(1 /a) / b = 1 / (a * b)`.
And the the `ceil(..)` represents the number of steps so it has to be multiplied by the step value.

**Comment by stefaniapedrazzi** [comment]
By the way I think it is better to use the complete name `basicTimeStep`.
I used step in my comment because it was shorter.
I'm sorry that it was not clear.

**Comment by DavidMansolino** [comment]
> No, (1 /a) / b = 1 / (a * b).

You are absolutely right, I did get confuse. Then maybe a simpler version would be:
codecogseqn 1

(Personnally keeping the (1000 / basicTimeStep) together helps because it represents the world refresh rate). Is this version simpler for you or you prefer:
codecogseqn 2

**Comment by stefaniapedrazzi** [comment]
Yes, your proposal is also good.

**Comment by DavidMansolino** [comment]
👍

**Pull request merged**